sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -0,0 +1,416 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace Azure.Messaging.EventHubs.Diagnostics
+{
+    /// <summary>
+    ///   Serves as an ETW event source for logging of information about
+    ///   Event Hubs client
+    /// </summary>
+    ///
+    /// <remarks>
+    ///   When defining Start/Stop tasks, it is highly recommended that the
+    ///   the StopEvent.Id must be exactly StartEvent.Id + 1.
+    /// </remarks>
+    ///
+    [EventSource(Name = "Azure-Messaging-EventHubs")]
```

Comment:
Brought forward from the track one code and updated for track two semantics.  Only a subset is used at present, but the pattern of implementation and logging areas were kept for consistency.